cmd/pack3d/main.go

```diff
@@ -40,80 +39,134 @@ func timed(name string) func() {
-		VolumeWithSpacing   float64
+		Filename          string
+		Transformation    [4][4]float64
+		VolumeWithSpacing float64
 	}
 
 	type err_msg struct {
-		Error            string
+		Error string
 	}
 
 	var (
 		singleStlSize []fauxgl.Vector
-	    done          func()
-	    totalVolume   float64
-		dimension     []float64
+		done          func()
+		totalVolume   float64
 		ntime         int
 		srcStlNames   []string
 		transMaps     []TransMap
-		)
+	)
 
 	rand.Seed(time.Now().UTC().UnixNano())
 
 	model := pack3d.NewModel()
 	count := 1
 	ok := false
 
-	// Loading build_volume size
-	for _, j := range os.Args[1:5]{
-		_dimension, err := strconv.ParseFloat(j, 64)
-		if err == nil{
-			dimension = append(dimension, float64(_dimension))
-			continue
-		}
-	}
-	spacing := dimension[3]/2.0
+	spacing := config.Spacing / 2.0
 	// frameSize is the vertex in the first quadrant
-	frameSize := fauxgl.V(dimension[0]/2.0, dimension[1]/2.0, dimension[2]/2.0)
-	buildVolume := dimension[0] * dimension[1] * dimension[2]
+	frameSize := fauxgl.V(config.BuildVolume[0]/2.0, config.BuildVolume[1]/2.0, config.BuildVolume[2]/2.0)
+	buildVolume := config.BuildVolume[0] * config.BuildVolume[1] * config.BuildVolume[2]
 	//fmt.Println(frameSize)
 
+	coPrintMap := make(map[string][]*Coprint)
 	/* Loading stl models */
-	for _, arg := range os.Args[6:] {
-		_count, err := strconv.ParseInt(arg, 0, 0)
-		if err == nil {
-			count = int(_count)
-			continue
-		}
+	for _, item := range config.Items {
+		done = timed(fmt.Sprintf("loading mesh %s", item.Filename))
+		var mesh *fauxgl.Mesh
+		var err error
+		if item.Coprint == nil {
+			mesh, err = fauxgl.LoadMesh(item.Filename)
+			if err != nil {
+				panic(err)
+			}
 
-		done = timed(fmt.Sprintf("loading mesh %s", arg))
-		mesh, err := fauxgl.LoadMesh(arg)
-		if err != nil {
-			panic(err)
-		}
-		done()
+			done()
 
-		totalVolume += mesh.BoundingBox().Volume()
-		size := mesh.BoundingBox().Size()
-		for i:=0; i<count; i++{
-			singleStlSize = append(singleStlSize, size)
-			srcStlNames = append(srcStlNames, arg)
-		}
+			totalVolume += mesh.BoundingBox().Volume()
+			size := mesh.BoundingBox().Size()
+			for i := 0; i < count; i++ {
+				singleStlSize = append(singleStlSize, size)
+				srcStlNames = append(srcStlNames, item.Filename)
+			}
 
-		fmt.Printf("  %d triangles\n", len(mesh.Triangles))
-		fmt.Printf("  %g x %g x %g\n", size.X, size.Y, size.Z)
+			fmt.Printf("  %d triangles\n", len(mesh.Triangles))
+			fmt.Printf("  %g x %g x %g\n", size.X, size.Y, size.Z)
+
+			done = timed("centering mesh")
+			mesh.Center()
+			done()
+		} else {
+			coPrintMap[item.Filename] = item.Coprint
+			mesh, err = fauxgl.LoadMesh(item.Filename)
+			if err != nil {
+				panic(err)
+			}
+			for _, cp := range item.Coprint {
+				coMesh, err := fauxgl.LoadMesh(cp.Filename)
+				if err != nil {
+					panic(err)
+				}
 
-		done = timed("centering mesh")
-		mesh.Center()
-		done()
+				coMesh.Transform(fauxgl.Matrix{
```

Comment:
This transform should be passed into `model.Add(mesh, bvhDetail, count, spacing, transform)` as an extra parameter.